### ISSUE TITLE
Ignoring exec() from previous page load.

### ISSUE THREAD

**Issue by Zeevac** [apache/cordova#266]
# Bug Report

## Problem
As you know, when orientation change on android platform, activity is killed and regenerated. Because of this, cordova platform is also regenerated. In this point, until cordova-android 9 version, all versions has executed the pluginInitialized method everytime when orientation changed but on cordova-android 9 it sometimes executes, sometimes not. When it is not called, log is prints below lines.
`
gap_init called from restricted origin:
D/CordovaBridge: Ignoring exec() from previous page load.
 ` 
After I searched for a solution on platform source code, I noticed that below code section is removed on cordova-android 9. When I readded this line, it works perfectly. Finally my question is that what issue prevent cordova calling pluginInitialize method every time when orientation changed?
### What is expected to happen?
When orientation is changed, pluginInitialize method on native must be called eveytime.

### What does actually happen?
It is called sometimes, sometimes not.

## Information

When I add below lines to the cordova.js file, everything works perfectly. These lines are removed from platform on cordova-android 9.0.0.
`
var utils = require('cordova/utils');
// Replace navigator before any modules are required(), to ensure it happens as soon as possible.
// We replace it so that properties that can't be clobbered can instead be overridden.
function replaceNavigator (origNavigator) {
    var CordovaNavigator = function () {};
    CordovaNavigator.prototype = origNavigator;
    var newNavigator = new CordovaNavigator();
    // This work-around really only applies to new APIs that are newer than Function.bind.
    // Without it, APIs such as getGamepads() break.
    if (CordovaNavigator.bind) {
        for (var key in origNavigator) {
            if (typeof origNavigator[key] === 'function') {
                newNavigator[key] = origNavigator[key].bind(origNavigator);
            } else {
                (function (k) {
                    utils.defineGetterSetter(newNavigator, key, function () {
                        return origNavigator[k];
                    });
                })(key);
            }
        }
    }
    return newNavigator;
}
`

### Command or Code

Just change orientation.

### Environment, Platform, Device

Windows 10
Cordova CLI 10.0.0
Cordova Android 9.0.0
Huawei P Smart Pro(probably all devices)

### Version information

❯ cordova -v
10.0.0
❯ cordova platform ls
Installed platforms:
  android 9.0.0

## Checklist

- [X] I searched for existing GitHub issues
- [X] I updated all Cordova tooling to most recent version
- [X] I included all the necessary information above

**Comment by raphinesse** [comment]
Thanks for reporting this issue and for the great information provided so far!

Apparently I am to blame for this: https://github.com/apache/cordova-js/pull/215. Now we just need to find out what exactly is happening and how to fix it.

Could it be that the JS-code you said you added back to `cordova.js` is missing the following lines:
```js
if (window.navigator) {
    window.navigator = replaceNavigator(window.navigator);
}
```

Without calling `replaceNavigator`, nothing should be happening at all... :thinking:

**Comment by raphinesse** [comment]
If you could provide a minimal app that lets us reproduce this issue, that would be very helpful.

**Comment by Zeevac** [comment]
> 
> 
> Thanks for reporting this issue and for the great information provided so far!
> 
> Apparently I am to blame for this: [apache/cordova-js#215](https://github.com/apache/cordova-js/pull/215). Now we just need to find out what exactly is happening and how to fix it.
> 
> Could it be that the JS-code you said you added back to `cordova.js` is missing the following lines:
> 
> ```js
> if (window.navigator) {
>     window.navigator = replaceNavigator(window.navigator);
> }
> ```
> 
> Without calling `replaceNavigator`, nothing should be happening at all... 🤔

Yes, I forgot to add these three last line while I was copying it :). Thank you for support.

**Comment by Zeevac** [comment]
> 
> 
> If you could provide a minimal app that lets us reproduce this issue, that would be very helpful.

https://github.com/Zeevac/myApp
Here it is a demo application to show issue. It will print 'blabla' to the console sometimes. But it has to print it everytime. If you try enough times, you will see that exec calls will be ignored by cordova. But if you add lines I mentioned in first message to the cordova.js, it works perfectly.

**Comment by Zeevac** [comment]
Hello @raphinesse. Is there any news?

**Comment by raphinesse** [comment]
Thanks for your feedback. There is no progress on my side yet. I'll look into it when I find the time. Unfortunately spare time is scarce for me at the moment.

If you or someone else wants to further help moving this forward, here's the next steps I would take:

- Find out how the navigator replacement code actually does anything.
  As described in the linked PR where I removed it, the assignment should not do anything per spec, as `window.navigator` is a read-only property. Thus `replaceNavigator` is probably modifying the `origNavigator` object in some non-obvious way.

- Understand how the missing changes in the navigator object break the JS/native bridge

-  Figure out how we want to fix this issue.
   Adding the old code back should not be the first choice, as it is misleading and not strict-mode compliant and we are eventually trying to get there.

**Comment by Zeevac** [comment]
> 
> 
> Thanks for your feedback. There is no progress on my side yet. I'll look into it when I find the time. Unfortunately spare time is scarce for me at the moment.
> 
> If you or someone else wants to further help moving this forward, here's the next steps I would take:
> 
>     * Find out how the navigator replacement code actually does anything.
>       As described in the linked PR where I removed it, the assignment should not do anything per spec, as `window.navigator` is a read-only property. Thus `replaceNavigator` is probably modifying the `origNavigator` object in some non-obvious way.
> 
>     * Understand how the missing changes in the navigator object break the JS/native bridge
> 
>     * Figure out how we want to fix this issue.
>       Adding the old code back should not be the first choice, as it is misleading and not strict-mode compliant and we are eventually trying to get there.

Any news? We are behind our schedule and it is very important to solve this issue.

**Comment by raphinesse** [comment]
Unfortunately there's no news on my side. Like many other I work on Cordova only in my spare time, which is rather scarce at the moment.

If this issue blocks your product, you have at least three options:
- Patch your local `cordova.js` (just re-insert the removed code) in a `before_prepare` [hook](https://cordova.apache.org/docs/en/9.x/guide/appdev/hooks/index.html)
- Try to move the progress of this issue along by working on the ToDos I posted above
- Hope that someone else resolves this issue

I understand that the lack of progress must be frustrating, but there's not much I can do right now.

**Comment by asv** [comment]
any status? very annoying bug.

**Comment by terreng** [comment]
I'm also seeing `Ignoring exec() from previous page load.` somewhat randomly. This is causing my app to essentially softlock because plugin code that I'm calling from javascript just isn't running.
